### ISSUE TITLE
dyld: lazy symbol binding failed: Symbol not found: _shmem_runtime_init

### ISSUE THREAD

**Issue by jeffhammond** [Sandia-OpenSHMEM/SOS#779]
Does SOS need to implement the equivalent of MPICH's `--enable-two-level-namespace`?

(This is on MacOS.)

# configure
```
./configure --prefix=/opt/shmem CC=gcc-8 FC=gfortran-8 CXX=g++-8 --with-ofi=/usr/local/Cellar/libfabric/1.6.1
```

# test fail
```
jrhammon-mac01:sandia-shmem jrhammon$ ./test/unit/pi
dyld: lazy symbol binding failed: Symbol not found: _shmem_runtime_init
  Referenced from: /Users/jrhammon/Work/PGAS/SHMEM/sandia-shmem/src/.libs/libsma.0.dylib
  Expected in: flat namespace

dyld: Symbol not found: _shmem_runtime_init
  Referenced from: /Users/jrhammon/Work/PGAS/SHMEM/sandia-shmem/src/.libs/libsma.0.dylib
  Expected in: flat namespace

Abort trap: 6
```

**Comment by jdinan** [comment]
This is falling through a hole in our configuration checks, allowing a build with no PMI client shim.

**Comment by jdinan** [comment]
Fixed by #783
